_posts/2019-12-09-reinvent-2019-recap.markdown

```diff
@@ -0,0 +1,130 @@
+<img src="/img/blog/reinvent-2019-recap/fargate-spot-instance.png" width="700">
+{: refdef}
+
+### SageMaker
+
+re:Invent 2019 had a lot to offer in the machine learning & AI space whith most notable entries coming in under the existing [SageMaker](https://aws.amazon.com/sagemaker/) suite.
+
+#### [SageMaker Studio](https://aws.amazon.com/blogs/aws/amazon-sagemaker-studio-the-first-fully-integrated-development-environment-for-machine-learning/)
+
+{:refdef: style="text-align: center;"}
+<img src="/img/blog/reinvent-2019-recap/reinvent-sagemaker-studio.png" width="700">
+{: refdef}
+
+SageMaker Studio was an exciting announcement as it marks a huge jump forward for the existing notebook platform AWS had to offer. SafeMaker Studio presents itself as a single pane of glass for all ML workflows, and it isn't lying about that.
+
+In the past I've had critisism for the lacklustre effort put into making SageMaker as useful as the competitions offerings (Google Notebooks, JupyterLab). Most of my personal problems came from:
+
+* No way to change instance type under notebook without shutting down experiments
+* Very lacking visualization / graphing capabilities
+* Model debugging & monitoring (couldn't easily run tensorboardX)
+
+Luckily all the capabilities above (and more) are supported in SageMaker Studio.
+
+#### [SageMaker AutoPilot](https://aws.amazon.com/blogs/aws/amazon-sagemaker-autopilot-fully-managed-automatic-machine-learning/)
+
+{:refdef: style="text-align: center;"}
+<img src="/img/blog/reinvent-2019-recap/reinvent-autopilot.png" width="700">
+{: refdef}
+
+Auto ML has been available on competing clouds for a while now ([Custom Vision AI](http://customvision.ai/), [Cloud AutoML](https://cloud.google.com/automl/)) so it was expected that AWS would be launching their own player in the ring.
+
+The idea behind **Auto Machine Learning** is that you are able to upload bulk data and with minimal direction on what features and characteristics you would like to see, the platform should be able to create a general model for you. It is usually very useful for quickly validating problems as it can setup some simple ML experiments for you without too much engineering work.
+
+SageMaker AutoPilot took this existing idea and (hopefully) has improved on it with a number of extra features like:
+
+* Automatic hyperparameter optimization
+* Easier distributed training
+* Better algorithm selection for different data types and problem scopes:
+  * Linear regression
+  * Binary classification
+  * Multi-class classification
+
+### [CodeGuru](https://aws.amazon.com/about-aws/whats-new/2019/12/aws-announces-amazon-codeguru-for-automated-code-reviews-and-application-performance-recommendations/)
+
+{:refdef: style="text-align: center;"}
+<img src="/img/blog/reinvent-2019-recap/reinvent-codeguru.png" width="700">
+{: refdef}
+
+CodeGuru is an automated; machine learning backed code review system that was built from internal code review data within AWS.
```

Comment:
replace the semicolon with a comma.
machine-learning-backed
add comma after backed.

```
CodeGuru is an automated, machine-learning-backed, code review system....